### PR TITLE
BUG/DEPR: Timestamp/Timedelta resolution

### PR DESCRIPTION
**Pull request by jbrockmendel**
Timedelta.resolution used to return a string, but that was deprecated in 0.25.0.  Enforce that deprecation to return Timedelta(nanoseconds=1), mirroring the stdlib timedelta.resolution behavior.

Timestamp.resolution correctly mirrors datetime.resolution by returning Timedelta(nanoseconds=1), but it is a property whereas for the stdlib it is a class attribute.  This PR makes it a class attribute for Timestamp.

### CODE REVIEW

File: pandas/tests/scalar/timedelta/test_timedelta.py [hunk 1]
```diff
@@ -804,9 +804,13 @@ def test_resolution_string(self):
     def test_resolution_deprecated(self):
```
Comment (mroeschke) [inline pandas/tests/scalar/timedelta/test_timedelta.py]: Maybe rename this test to  just `test_resolution`

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 2]
```diff
@@ -1602,3 +1552,4 @@ cdef _broadcast_floordiv_td64(int64_t value, object other,
 # resolution in ns
 Timedelta.min = Timedelta(np.iinfo(np.int64).min + 1)
 Timedelta.max = Timedelta(np.iinfo(np.int64).max)
+Timedelta.resolution = Timedelta(nanoseconds=1)
```
Comment (mroeschke) [inline pandas/_libs/tslibs/timedeltas.pyx]: Think a newline is needed here.

**Review by WillAyd — APPROVED** [review]
lgtm ex @mroeschke comments

**Comment by jbrockmendel** [comment]
updated+green

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**